runtime/doc/lua-guide.txt

```diff
@@ -0,0 +1,758 @@
+Options                                                      *lua-guide-options*
+
+There are two complementary ways of setting |options| via Lua.
+
+------------------------------------------------------------------------------
+vim.opt
+
+The most convenient way for setting global and local options, e.g., in `init.lua`,
+is through `vim.opt` and friends:
+
+• |vim.opt|        - behaves like |:set|
+• |vim.opt_global| - behaves like |:setglobal|
+• |vim.opt_local|  - behaves like |:setlocal|
+
+For example, the Vimscript commands
+>vim
+    set smarttab
+    set nosmarttab
+<
+are equivalent to
+>lua
+    vim.opt.smarttab = true
+    vim.opt.smarttab = false
+<
+In particular, they allow an easy way to working with list-like, map-like, and
+set-like options through Lua tables: Instead of
+>vim
+    set wildignore=*.o,*.a,__pycache__
+    set listchars=space:_,tab:>~
+    set formatoptions=njt
+<
+you can use
+>lua
+    vim.opt.wildignore = { '*.o', '*.a', '__pycache__' }
+    vim.opt.listchars = { space = '_', tab = '>~' }
+    vim.opt.formatoptions = { n = true, j = true, t = true }
+<
+These wrappers also come with methods that work similarly to their |:set+=|,
+|:set^=| and |:set-=| counterparts in Vimscript:
+>lua
+    vim.opt.shortmess:append({ I = true })
+    vim.opt.wildignore:prepend('*.o')
+    vim.opt.whichwrap:remove({ 'b', 's' })
+<
+The price to pay is that you cannot access the option values directly but must
+use |vim.opt:get()|:
+>lua
+    print(vim.opt.smarttab)
+    --> {...} (big table)
+    print(vim.opt.smarttab:get())
+    --> false
+    vim.pretty_print(vim.opt.listchars:get())
+    --> { space = '_', tab = '>~' }
+<
+------------------------------------------------------------------------------
+vim.o
+
+For this reason, there exists a more direct variable-like access using `vim.o`
+and friends, similarly to how you can get and set options via `:echo &number`
+and `:let &listchars='space:_,tab:>~'`:
+
+• |vim.o|  - behaves like |:set|
+• |vim.go| - behaves like |:setglobal|
+• |vim.bo| - for buffer-scoped options
+• |vim.wo| - for window-scoped options
+
+For example:
+>lua
+    vim.o.smarttab = false -- :set nosmarttab
+    print(vim.o.smarttab)
+    --> false
+    vim.o.listchars = 'space:_,tab:>~' -- :set listchars='space:_,tab:>~'
+    print(vim.o.listchars)
+    --> 'space:_,tab:>~'
+    vim.o.isfname = vim.o.isfname .. ',@-@' -- :set isfname+=@-@
+    print(vim.o.isfname)
+    --> '@,48-57,/,.,-,_,+,,,#,$,%,~,=,@-@'
+    vim.bo.shiftwidth = 4 -- :setlocal shiftwidth=4
+    print(vim.bo.shiftwidth)
+    --> 4
+<
+Just like variables, you can specify a buffer number or |window-ID| for buffer
+and window options, respectively. If no number is given, the current buffer or
+window is used:
+>lua
+    vim.bo[4].expandtab = true -- sets expandtab to true in buffer 4
+    vim.wo.number = true       -- sets number to true in current window
+    print(vim.wo[0].number)    --> true < Note: |vim.wo| only sets
```

Comment:
only sets what?

Reply:
Oops, I seem to have messed up a rebase! (Below as well.)

Reply:
Originally, that was going to be a note about window-scoped local options and a reference to `nvim_set_option_value`. I decided against it in the end.